### ISSUE TITLE
REQUEST: Controlling the LED/Light

### ISSUE THREAD

**Issue by Mazecreator** [zxing-js/library#267]
It would be nice to be able to activate/deactivate the light on the camera.  This would be a great feature to be able to use this in poor lighting conditions.

**Comment by slokhorst** [comment]
This sounds very useful! Here is a code sample to start out with:
https://stackoverflow.com/a/47153547/1427212

**Comment by odahcam** [comment]
We actually already did that! 🎉 
But it's only available on our Angular wrapper. 😭

Just take a look on the code (just follow the torch haha): 

https://github.com/zxing-js/ngx-scanner/blob/1079dec0e905821fa4e3038a4ced095e395f1a64/projects/zxing-scanner/src/lib/browser-multi-format-continuous-reader.ts#L129-L144

So, any of you would be very welcome to port the code from there to here so we could make use of that nice feature here too. 😊

Thanks.

**Comment by slokhorst** [comment]
It's actually pretty easy to use with the library currently:

```js
/* existing code to setup and start ZXing */
const scanner = new ZXing.BrowserMultiFormatReader();
scanner.decodeFromVideoDevice(/* ... */);

/* now the addition to start the torch */
scanner.stream.getVideoTracks()[0].applyConstraints({
    advanced: [{torch: torchStatus}]
});
```

**Comment by sabitertan** [comment]
@andretgregorio Didn't test, but you are not passing current device Id to decodeFromVideoDevice.

**Comment by odahcam** [comment]
@sabitertan thanks for sharing, very interesting [torch function](https://github.com/sabitertan/BlazorBarcodeScanner/blob/d321d5f58cedffba60c27fe3161b2319760fcf76/BlazorBarcodeScanner.ZXing.JS/wwwroot/BlazorBarcodeScanner.js#L2) indeed. I played a lot with torch myself and noticed a lot of inconsistencies between devices and browsers, did you notice it too? If so, how are you working around it?

**Comment by sabitertan** [comment]
@odahcam I initially get that function from[ browser implementation](https://github.com/zxing-js/browser/blob/master/src/readers/BrowserCodeReader.ts#L63) :)
I just implemented torch but I think there is no way turn "off" unless you destroy stream/track and reinitialize it.
(at least that is what I understand from standard :) https://w3c.github.io/mediacapture-image/#torch )

And yes still inconsistence API, needs more testing :/

**Comment by odahcam** [comment]
Yeah, it's kinda tricky and painful hahah. So I discovered some controls work better than others and in some devices you really can't turn off the torch and in some you actually can. Very odd, but at the same time that's kind of we can expect from an experimental bleeding edge API...
